src/js/scope.ts

```diff
@@ -59,12 +60,22 @@ export class ReactNativeScope extends Scope {
    * @inheritDoc
    */
   public addBreadcrumb(breadcrumb: Breadcrumb, maxBreadcrumbs?: number): this {
-    const mergedBreadcrumb = {
+    const mergedBreadcrumb: Breadcrumb = {
       ...breadcrumb,
       level: breadcrumb.level || DEFAULT_BREADCRUMB_LEVEL,
+      data: breadcrumb.data ? convertToNormalizedObject(breadcrumb.data) : undefined,
     };
-    NATIVE.addBreadcrumb(mergedBreadcrumb);
-    return super.addBreadcrumb(mergedBreadcrumb, maxBreadcrumbs);
+
+    const countBefore = this._breadcrumbs.length;
+    super.addBreadcrumb(mergedBreadcrumb, maxBreadcrumbs);
+    const countAfter = this._breadcrumbs.length;
+
+    const breadcrumbAdded = countAfter > countBefore;
+    if (breadcrumbAdded) {
```

Comment:
I don't follow why we need the `breadcrumbAdded` condition and `countBefore`/`countAfter`, mind elaborating?

Reply:
Thanks for the catch, it's not needed. The oldest breadcrumb gets removed if the max is reached.